source/MRMesh/MRMeshLoadStep.cpp

```diff
@@ -15,62 +19,102 @@
 #endif
 #include <opencascade/BRep_Tool.hxx>
 #include <opencascade/BRepMesh_IncrementalMesh.hxx>
+#include <opencascade/Message.hxx>
+#include <opencascade/Message_Printer.hxx>
+#include <opencascade/Message_PrinterOStream.hxx>
 #include <opencascade/STEPControl_Reader.hxx>
+#include <opencascade/StepData_Protocol.hxx>
+#include <opencascade/StepData_StepModel.hxx>
+#include <opencascade/StepData_StepWriter.hxx>
 #include <opencascade/TopExp_Explorer.hxx>
 #include <opencascade/TopoDS.hxx>
 #pragma warning( pop )
 
-namespace MR::MeshLoad
+namespace
 {
 
-Expected<Mesh, std::string> fromStep( const std::filesystem::path& path, VertColors* colors, ProgressCallback callback )
-{
-    std::ifstream in( path, std::ifstream::binary );
-    if ( !in )
-        return unexpected( std::string( "Cannot open file for reading " ) + utf8string( path ) );
+using namespace MR;
 
-    return addFileNameInError( fromStep( in, colors, callback ), path );
-}
-
-Expected<Mesh, std::string> fromStep( std::istream& in, VertColors*, ProgressCallback callback )
+class SpdlogPrinter final : public Message_Printer
 {
-    MR_TIMER
+private:
+    void send( const TCollection_AsciiString& string, const Message_Gravity gravity ) const override
+    {
+        auto level = spdlog::level::trace;
+        switch ( gravity )
+        {
+            case Message_Trace:
+                level = spdlog::level::trace;
+                break;
+            case Message_Info:
+                level = spdlog::level::info;
+                break;
+            case Message_Warning:
+                level = spdlog::level::warn;
+                break;
+            case Message_Alarm:
+                level = spdlog::level::err;
+                break;
+            case Message_Fail:
+                level = spdlog::level::critical;
+                break;
+        }
 
-    // NOTE: OpenCASCADE STEP reader is NOT thread-safe
-    static std::mutex mutex;
-    std::unique_lock lock( mutex );
+        spdlog::log( level, "OpenCASCADE: {}", string.ToCString() );
+    }
+};
 
-    STEPControl_Reader reader;
+class MessageHandler
+{
+public:
+    MessageHandler()
+        : printer_( new SpdlogPrinter )
     {
-        MR_NAMED_TIMER( "STEP reader: read stream" )
-        reader.ReadStream( "STEP file", in );
+        auto& messenger = Message::DefaultMessenger();
+        // remove default stdout output
+        messenger->RemovePrinters( STANDARD_TYPE( Message_PrinterOStream ) );
+        // add spdlog output
+        messenger->AddPrinter( Handle( Message_Printer)( printer_ ) );
     }
+
+private:
+    SpdlogPrinter* printer_;
```

Comment:
let us store it by value `SpdlogPrinter printer_;` and implement the destructor `~MessageHandler()` that will Remove the printer